### DISCUSSION TITLE
Why some methods call handlers and others don't?

### DISCUSSION THREAD

**Discussion by damccull** [ravendb/ravendb#14290] [Q&A] [answered]
While do these two methods _not_ call the `beforeDelete` listener
```csharp
void Delete(string id, string expectedChangeVector);
void Delete(string id);
```

but this one does?

```csharp
void Delete<T>(T entity);
```

Are there any others for the same reason?

**Comment by ayende** [comment, marked as answer]
We do call that on `Delete<T>(entity)`.  This is done here: 

https://github.com/ravendb/ravendb/blob/0fdeb6dd517b69dc7aef89bbd5bebc1f6e80393d/src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs#L1037

**Reply by damccull** [reply]
Thanks for the quick reply. I do know it's called on that one. My question is why is it not called on the other two?

**Reply by ayende** [reply]
It is called, just not in the same location. See:
https://github.com/ravendb/ravendb/blob/4e1478e2b6f0b298e0432d588f347b8a2f606730/src/Raven.Client/Documents/Commands/Batches/DeleteCommandData.cs#L48

**Reply by damccull** [reply]
Oh ok. Cool. I was asking because the doc comments have a warning that those two don't call it and I was wondering why they might not, like if there was some API reason for it.

**Reply by ayende** [reply]
Those are docs bugs, sorry, we'll fix them.

**Reply by ayende** [reply]
https://issues.hibernatingrhinos.com/issue/RavenDB-18704

**Reply by damccull** [reply]
Thanks for the clarification!
